File: algorithm/plusMinus/answer.js
function plusMinus(arr) {
  let [posCount, negCount, neuCount] = [0, 0, 0];

  for (let i = 0; i < arr.length; i++) {
    let number = arr[i];
    if (number > 0) posCount += 1;
    if (number < 0) negCount += 1;
    if (number === 0) neuCount += 1;
  }

  const posR = posCount / arr.length;
  const negR = negCount / arr.length;
  const neuR = neuCount / arr.length;
  console.log(posR.toFixed(6));
  console.log(negR.toFixed(6));
  console.log(neuR.toFixed(6));
}

File: algorithm/plusMinus/answer.java

import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {
  
    static String fixedDecimal(float value, int to) {
      return String.format("%." + to + "f", value);
    }

    static void plusMinus(int[] arr) {
      int arrLength = arr.length ;
      float posCount = 0;
      float negCount = 0;
      float neuCount = 0;

      for (int i = 0; i < arrLength; i++) {
        int value = arr[i];
        
        if (value > 0) posCount++;
        else if (value < 0) negCount++;
        else if (value == 0) neuCount++;
      }
      
      float posRatio = posCount / arrLength;
      float negRatio = negCount / arrLength;
      float neuRatio = neuCount / arrLength;
      System.out.println(fixedDecimal(posRatio, 6));
      System.out.println(fixedDecimal(negRatio, 6));
      System.out.println(fixedDecimal(neuRatio, 6));
    }
}
